web/modules/custom/search_api_field/src/Plugin/facets/facet_source/SearchApiFieldDeriver.php

```diff
@@ -61,6 +61,7 @@ public static function create(ContainerInterface $container, $base_plugin_id) {
    * {@inheritdoc}
    */
   public function getDerivativeDefinitions($base_plugin_definition) {
+    $plugin_derivatives = [];
```

Comment:
This should be defined in its scope, meaning outside the foreach below. There is no reason to initialize all variables in the top of the method.